model/daal4py/dffml_model_daal4py/daal4pylr.py

```diff
@@ -0,0 +1,101 @@
+import pathlib
+import statistics
+import importlib
+import hashlib
+import os
+
+from typing import AsyncIterator, Tuple, Any, Type, List
+
+from dffml import (
+    config,
+    field,
+    entrypoint,
+    SimpleModel,
+    ModelNotTrained,
+    Accuracy,
+    Feature,
+    Features,
+    Sources,
+    Record,
+)
+
+
+@config
+class DAAL4PyLRModelConfig:
+    predict: Feature = field("Label or the value to be predicted")
+    features: Features = field("Features to train on. For SLR only 1 allowed")
+    directory: pathlib.Path = field(
+        "Directory where state should be saved",
+        default=pathlib.Path("~", ".cache", "dffml", "daal4py"),
+    )
+
+
+@entrypoint("daal4pylr")
+class DAAL4PyLRModel(SimpleModel):
+    CONFIG: DAAL4PyLRModelConfig
+
+    def __init__(self, parent) -> None:
+        super().__init__(parent)
+        self.d4p = importlib.import_module("daal4py")
+        self.pd = importlib.import_module("pandas")
+        self.np = importlib.import_module("numpy")
+        self.features = self.applicable_features
+        self._features_hash = self._feature_predict_hash()
+        self.lm = self.d4p.linear_regression_training(interceptFlag=True, streaming = True)
+        self.lm_predictor = d4p.linear_regression_prediction(streaming = True)
+        self.path = self.filepath(self.parent.config.directory, "trained_model.sav")
+        self.lm_trained = None
+        load_model()
+
+    def load_model(self):
+        if self.path.is_file():
+            self.lm_trained = self.joblib.load(self.path)
+
+    def filepath(self, directory, file_name):
+        return os.path.join(directory, file_name)
```

Comment:
In `load_model` you use `.is_file()`, but here you take `directory` which is an instance of `pathlib.Path`, and you pass it to `os.path.join` which will return a string, rather than a `pathlib.Path` instance, which means the return value of `filepath` is a string, which means that `self.path` is a string, so it won't have a `.is_file()` method. We should instead use [`pathlib.Path.joinpath`](https://docs.python.org/3/library/pathlib.html?highlight=pathlib%20path#pathlib.PurePath.joinpath) or the [`/` operator](https://docs.python.org/3/library/pathlib.html?highlight=pathlib%20path#operators)

Reply:
Haven't played a lot with paths. It makes a lot of sense. Thanks a bunch!